packages/wp-dashboard/src/api/settings.js

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { addQueryArgs } from '@web-stories-wp/design-system';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Fetch settings.
+ * Used on settings page.
+ *
+ * @param {string} apiPath API path.
+ * @return {Promise} Request promise.
+ */
+export function fetchSettings(apiPath) {
+  return apiFetch({
+    path: apiPath,
+  }).then((response) => ({
+    googleAnalyticsId: response.web_stories_ga_tracking_id,
+    usingLegacyAnalytics: response.web_stories_using_legacy_analytics,
+    adSensePublisherId: response.web_stories_adsense_publisher_id,
+    adSenseSlotId: response.web_stories_adsense_slot_id,
+    adManagerSlotId: response.web_stories_ad_manager_slot_id,
+    adNetwork: response.web_stories_ad_network,
+    videoCache: response.web_stories_video_cache,
+    archive: response.web_stories_archive,
+    archivePageId: response.web_stories_archive_page_id,
+  }));
```

Comment:
```suggestion
  }).then((response) => ({
    googleAnalyticsId: response.web_stories_ga_tracking_id,
    usingLegacyAnalytics: response.web_stories_using_legacy_analytics,
    adSensePublisherId: response.web_stories_adsense_publisher_id,
    adSenseSlotId: response.web_stories_adsense_slot_id,
    adManagerSlotId: response.web_stories_ad_manager_slot_id,
    adNetwork: response.web_stories_ad_network,
    videoCache: response.web_stories_video_cache,
    archive: response.web_stories_archive,
    archivePageId: response.web_stories_archive_page_id,
  }));
```

This logic is repeated. Work making it's own function.

Reply:
Done in https://github.com/google/web-stories-wp/pull/9237/commits/fe81622d0a1ab8dd66dab337f0e8f2ee43afb20a